internal/handlers/pg/msg_find.go

```diff
@@ -104,6 +104,9 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 			return lazyerrors.Error(err)
 		}
 
+		// SortIterator should be closed
+		defer iter.Close()
+
```

Comment:
Should we  `defer iter.Close()` on `ProjectionIterator` few lines below too?

Reply:
> // Close method closes the underlying iterator.
> // For that reason, there is no need to track both iterators.

There are two types of iterators: ones that wrap some resources (slices of data, database connections, etc) that should be freed and ones that wrap other iterators. The latter "chain" to each other, and only one (any) of them should be closed.

In retrospect, that's a leaky abstraction and a design mistake. We should think about fixing it

Reply:
Okay, so ProjectionIterator doesn't need resources and only SortIterator need resources, so it's fine to not close it here I see.